Aspire Budgeting/Facilities/StateManager.swift

```diff
@@ -146,11 +121,56 @@ final class StateManager: ObservableObject {
     }
   }
 
-  func canTransition(to nextState: State) -> Bool {
-    guard let validTransitions = transitions[currentState] else {
+  func canTransition(to nextState: AppState) -> Bool {
+    guard let validTransitions = transitions[currentState.value] else {
       return false
     }
 
     return validTransitions.contains(nextState)
   }
 }
+
+// MARK: - Computed Properties
+extension StateManager {
+  var needsLocalAuth: Bool {
+    let currentValue = currentState.value
+    return currentValue == .verifiedExternally
+      || currentValue == .localAuthFailed
+      || currentValue == .needsLocalAuthentication
+  }
+
+  var isLoggedOut: Bool {
+    let currentValue = currentState.value
+    return currentValue == .loggedOut
+  }
+
+  var hasDefaultSheet: Bool {
+    let currentValue = currentState.value
+    return currentValue == .hasDefaultSheet
+  }
+}
+
+//MARK: - AppStateManager Protocol Methods
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/aspirebudgetingmobile/aspirebudgeting_ios/blob/0064a2de7513bf35f3612cdd6552d081d2b41cdd/Aspire Budgeting/Facilities/StateManager.swift#L153"></span>MARK comment should be in valid format. e.g. ‘// MARK: …’ or ‘// MARK: - …’<br />
<code>mark</code> <code>StateManager.swift:153</code></td>
    </tr>
  </tbody>
</table>